pkg/tar/tar_test.go

```diff
@@ -185,23 +187,60 @@ func TestCreateTar(t *testing.T) {
 	verifyTarFile(t, tarFile, testFiles, testLinks)
 }
 
-func createTestTar(files []fileDesc, writer io.Writer) {
+func createTestTar(files []fileDesc, writer io.Writer, t *testing.T) {
```

Comment:
It would be more idiomatic if, instead of passing `t *testing.T` as an argument to `createTestTar` and have `createTestTar` call `t.Fatal*`, `createTestTar` would return an error and the caller (`Test*` function) handles the error and call `t.Fatal*`/`t.Error*` as appropriate.